graylog2-server/src/main/java/org/graylog2/streams/StreamRouterEngine.java

```diff
@@ -203,6 +207,13 @@ public String getFingerprint() {
 
         for (Stream stream : result) {
             streamMetrics.markIncomingMeter(stream.getId());
+            if(stream.getRemoveFromAllMessages()) {
+                if(message.removeStream(defaultStream)) {
+                    LOG.trace("Successfully removed default stream <{}> from message <{}>", defaultStream.getId(), message.getId());
```

Comment:
I would wrap this in `isTraceEnabled` to avoid the `getId()` calls for the stream and message objects for every message.